packages/editor/src/lib/editor/types/misc-types.ts

```diff
@@ -55,6 +55,18 @@ export interface TLCameraOptions {
 	constraints?: TLCameraConstraints
 }
 
+/** @public */
+export interface AssetContextProps {
+	zoom: number
```

Comment:
instead of passing `zoom`, could we pass something like `screenScale`? This would be the page-space scale factor of the image (ie how much it has been resized from its original in page space) multiplied by the zoom level to produce how much it's resized from its original in screen space.

I think right now, we don't take into account the scaling of the image at all

Reply:
i'm gonna ask a silly question: how does one tell how much something has been resized? (original vs. current size)
i feel like i'm not seeing how to do that 🤔

Reply:
image shapes have a width/height property (how big is the shape) which is separate from the width/height of the asset (the natural pixel dimensions of the image). The image shape would need to tell the hook what size it's expecting to render on the page so it can combine it with the page transform

Reply:
ah right right, ok - for some reason i was conflating original shape size and asset size.
i think the odd thing, slightly, is that the asset does get resized on initial load which, maybe we should rethink?

Reply:
i think i wasn't clear in that last comment - what i mean to say is, maybe we nix this piece of code: https://github.com/tldraw/tldraw/blob/main/packages/tldraw/src/lib/defaultExternalContentHandlers.ts#L84-L90

the clamping is fine - but i'm wondering if it should be set to 1.0 for quality instead of 0.92 🤔

Reply:
So that quality param i think is a bit different to cloudflare's one. It's the format-quality for lossy formats like jpeg and webp: https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob#quality

i think the ideal thing to do here would be to check what the original quality was somehow and re-use that. for lossy formats, setting quality to 1 can cause a bit of an explosion in file size - for a random photo jpeg in my downloads, setting quality from whatever it is right now to 1 causes the file size to go from 2.3mb to 7.6mb

i do agree that we could arguably maybe nix that now though?

Reply:
nixed 👍